plugin/src/main/java/org/owasp/benchmarkutils/score/parsers/GitLabSastReader.java

```diff
@@ -0,0 +1,108 @@
+package org.owasp.benchmarkutils.score.parsers;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.owasp.benchmarkutils.score.*;
+
+public class GitLabSastReader extends Reader {
+    @Override
+    public boolean canRead(ResultFile resultFile) {
+        return resultFile.isJson()
+                && resultFile.json().has("scan")
+                && resultFile
+                .json()
+                .getJSONObject("scan")
+                .getJSONObject("analyzer")
+                .getJSONObject("vendor")
+                .getString("name")
+                .equalsIgnoreCase("GitLab");
+    }
+
+    @Override
+    public TestSuiteResults parse(ResultFile resultFile) throws Exception {
+        TestSuiteResults tr = new TestSuiteResults("GitLab-SAST", true, TestSuiteResults.ToolType.SAST);
+
+        JSONArray vulnerabilities = resultFile.json().getJSONArray("vulnerabilities");
+
+        for (int vulnerability = 0; vulnerability < vulnerabilities.length(); vulnerability++) {
+            TestCaseResult tcr = parseGitLabSastFindings(vulnerabilities.getJSONObject(vulnerability));
+            if (tcr != null) {
+                tr.put(tcr);
+            }
+        }
+        return tr;
+    }
+
+    private TestCaseResult parseGitLabSastFindings(JSONObject vulnerability) {
+
+        try {
+            String className = vulnerability.getJSONObject("location").getString("file");
+            className = (className.substring(className.lastIndexOf('/') + 1)).split("\\.")[0];
+
+            if (className.startsWith(BenchmarkScore.TESTCASENAME)) {
```

Comment:
Can you use Reader#testNumber and pass `BenchmarkScore.TESTCASENAME` to it? So instead of

```
String className = vulnerability.getJSONObject("location").getString("file");
className = (className.substring(className.lastIndexOf('/') + 1)).split("\\.")[0];

if (className.startsWith(BenchmarkScore.TESTCASENAME)) {
[...]
tcr.setNumber(testNumber(className));
```

Directly check for the test number:

```
int testNumber = testNumber(vulnerability.getJSONObject("location").getString("file"));

if(testNumber > -1) {
[...]
tcr.setNumber(testNumber);
```

Just an idea to reduce the amount of code. Disclaimer: Untested, just an idea 😅

Reply:
Thanks, Modified it and it is working

Reply:
Great, thank you!